queue/driver_mongo.go

```diff
@@ -677,24 +677,40 @@ func (d *mongoDriver) GetAllAttempts(ctx context.Context, id string) ([]amboy.Jo
 }
 
 func (d *mongoDriver) Put(ctx context.Context, j amboy.Job) error {
-	ji, err := registry.MakeJobInterchange(j, d.opts.Format)
-	if err != nil {
-		return errors.Wrap(err, "converting in-memory job to interchange job")
+	return d.PutMany(ctx, []amboy.Job{j})
+}
+
+// PutMany enqueues multiple jobs on the queue.
+// Returns an error if any of the jobs cannot be inserted. If a non-duplicate-job error
+// is encountered it is returned. If every job is a duplicate job error then if any
+// of them is a duplicate scope error a duplicate scope error is returned.
```

Comment:
What if we used a catcher to add both duplicate scope and duplicate write errors?

Reply:
We can't use a catcher because the catcher destroys all the error types and merges them into one untyped error. This makes it difficult to determine individual jobs errors (for things like `EnqueueUniqueJobs`) if the jobs errored due to all duplicate IDs, all duplicate scopes, a mix of duplicate + other DB errors, etc.

Reply:
As long as the error priority is clear here, I think that is fine then.